lib/ansible/modules/identity/keycloak/keycloak_user.py

```diff
@@ -0,0 +1,471 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Nicolas Duclert <nicolas.duclert@metronlab.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'], 'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: keycloak_user
+
+short_description: Allows administration of Keycloak users via Keycloak API
+
+version_added: "2.10"
+
+description:
+    - This module allows the administration of Keycloak clients via the Keycloak REST API. It
+      requires access to the REST API via OpenID Connect; the user connecting and the client being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate client definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/4.8/rest-api/index.html/).
+      Aliases are provided so camelCased versions can be used as well. If they are in conflict
+      with ansible names or previous used names, they will be prefixed by "keycloak".
+
+options:
+    state:
+        description:
+            - State of the user
+            - On C(present), the user will be created (or updated if it exists already).
+            - On C(absent), the user will be removed if it exists
+        choices: [ present, absent ]
+        default: present
+        type: str
+
+    realm:
+        description:
+            - The realm to create the user in.
+        default: master
+        type: str
+
+    attributes:
+        description:
+            - a dictionary with the key and the value to put in keycloak.
+              Keycloak will always return the value in a list of one element.
+              Keys and values are converted into string.
+        required: false
```

Comment:
Required: false is implicit and not required

Reply:
Changed in 525d5f2121c0558c972ffde3bae5fb3b2a29a9c1.